### ISSUE TITLE
pointwise scheduler issue on devel

### ISSUE THREAD

**Issue by jjsjann123** [csarofeen/pytorch#2053]
### 🐛 Describe the bug

I'm still working on a repro. Currently hit this on my side branch where I have upstream master merged into devel.

Hitting this assert:
```
RuntimeError: rhs_i >= 0 && lhs_i >= 0 INTERNAL ASSERT FAILED at "/opt/pytorch/pytorch/torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp":621, please report a bug to PyTorch.
```

Here's the fusion definition
```
def nvfuser_fusion_id44(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[1, -1, -1, -1], contiguous=[True, True, True, True], dtype=DataType.Bool)
    T1 = fd.define_tensor(symbolic_sizes=[1, -1, 1, 1], contiguous=[True, True, True, True], dtype=DataType.Bool)
    T2 = fd.define_tensor(symbolic_sizes=[1, -1, -1, -1], contiguous=[False, False, False, True], dtype=DataType.Float)
    T3 = fd.define_tensor(symbolic_sizes=[1, -1, -1, -1], contiguous=[True, True, True, True], dtype=DataType.Float)
    T4 = fd.ops.cast(T0, dtype=DataType.Float)
    T5 = fd.ops.broadcast_in_dim(T1, output_shape=[1, 1024, 12, 513], broadcast_dims=[0, 1, 2, 3])
    S6 = fd.define_constant(1.1111111111111112)
    T7 = fd.ops.mul(T4, S6)
    T8 = fd.ops.mul(T2, T7)
    S9 = fd.define_constant(0.0000000000000000)
    T10 = fd.ops.where(T5, S9, T8)
    T11 = fd.ops.mul(T10, T3)
    T12 = fd.ops.sum(T11, axes=[3], keepdim=False, dtype=DataType.Null)
    T13 = fd.ops.broadcast_in_dim(T12, output_shape=[1, 1024, 12, 1], broadcast_dims=[0, 1, 2])
    T14 = fd.ops.broadcast_in_dim(T13, output_shape=[1, 1024, 12, 513], broadcast_dims=[0, 1, 2, 3])
    T15 = fd.ops.mul(T3, T14)
    T16 = fd.ops.sub(T11, T15)
    T17 = fd.squeeze(T16, original_shape=[1, 1024, 12, 513], dim=0)
    fd.add_output(T17)
```

Which gives us a segmented fusion:
```
Segment the fusion (Original Fusion Un-modified):
Inputs:
  T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ], bool
  T1_g[ bS4{1}, iS5{i7}, bS6{1}, bS7{1} ], bool
  T2_g[ bS8{1}, iS9{i11}, iS10{i12}, iS11{i13} ], float
  T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ], float
Outputs:
  T17_g[ rS68{1}, iS69{i7}, iS70{i12}, iS71{i13} ], float

%kernel_math {
T5_l[ bS20{1}, iS21{i7}, bS22{1}, bS23{1} ]
   = T1_g[ bS4{1}, iS5{i7}, bS6{1}, bS7{1} ];
T6_g[ bS24{1}, iS25{i7}, bS26{12}, bS27{513} ] = expand( T5_l[ bS20{1}, iS21{i7}, bS22{1}, bS23{1} ], {1, i7, 12, 513} )
T4_l[ bS16{1}, iS17{i1}, iS18{i2}, iS19{i3} ]
   = (float)(T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ]);
T7_g[ bS28{1}, iS29{i1}, iS30{i2}, iS31{i3} ]
   = T4_l[ bS16{1}, iS17{i1}, iS18{i2}, iS19{i3} ]
   * double(1.1111111111111112);
T8_g[ bS32{1}, iS33{i11}, iS34{i12}, iS35{i13} ]
   = T2_g[ bS8{1}, iS9{i11}, iS10{i12}, iS11{i13} ]
   * T7_g[ bS28{1}, iS29{i1}, iS30{i2}, iS31{i3} ];
T9_l[ bS36{1}, iS37{i7}, iS38{i12}, iS39{i13} ]
   = where(T6_g[ bS24{1}, iS25{i7}, bS26{12}, bS27{513} ]
  , double(0.0000000000000000)
  , T8_g[ bS32{1}, iS33{i11}, iS34{i12}, iS35{i13} ]);
T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ]
   = T9_l[ bS36{1}, iS37{i7}, iS38{i12}, iS39{i13} ]
   * T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ];
T11_g[ bS44{1}, iS45{i7}, iS46{i12}, rS47{i13} ]
   = reduction( T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ], op = add, initial value = double(0.0000000000000000), allreduce = false )
T12_l[ bS48{1}, iS49{i7}, iS50{i12}, bS51{1} ]
   = broadcast( T11_g[ bS44{1}, iS45{i7}, iS46{i12}, rS47{i13} ] )
T13_g[ bS52{1}, iS53{i7}, iS54{i12}, bS55{1} ]
   = T12_l[ bS48{1}, iS49{i7}, iS50{i12}, bS51{1} ];
T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ] = expand( T13_g[ bS52{1}, iS53{i7}, iS54{i12}, bS55{1} ], {1, i7, i12, 513} )
T15_l[ bS60{1}, iS61{i15}, iS62{i16}, iS63{i17} ]
   = T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ]
   * T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ];
T16_g[ bS64{1}, iS65{i7}, iS66{i12}, iS67{i13} ]
   = T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ]
   - T15_l[ bS60{1}, iS61{i15}, iS62{i16}, iS63{i17} ];
T17_g[ rS68{1}, iS69{i7}, iS70{i12}, iS71{i13} ]
   = reduction( T16_g[ bS64{1}, iS65{i7}, iS66{i12}, iS67{i13} ], op = add, initial value = double(0.0000000000000000), allreduce = false )
} 
```

```
Segmented_Fusion Dump: -- fusion segments:
Segmented_Fusion{
groups:
g{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}

g{11, 12, 13}

edges:
e{ g{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
 -> g{11, 12, 13}
(T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ]) }

e{ g{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
 -> g{11, 12, 13}
(T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ]) }


group details:
g{(reduction)
inputs:
T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ] bool
T1_g[ bS4{1}, iS5{i7}, bS6{1}, bS7{1} ] bool
T2_g[ bS8{1}, iS9{i11}, iS10{i12}, iS11{i13} ] float
T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ] float
i7 int64_t
i12 int64_t
outputs:
T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ] float
T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ] float

T4_l[ bS16{1}, iS17{i1}, iS18{i2}, iS19{i3} ]
   = (float)(T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ]);
T7_g[ bS28{1}, iS29{i1}, iS30{i2}, iS31{i3} ]
   = T4_l[ bS16{1}, iS17{i1}, iS18{i2}, iS19{i3} ]
   * double(1.1111111111111112);
T8_g[ bS32{1}, iS33{i11}, iS34{i12}, iS35{i13} ]
   = T2_g[ bS8{1}, iS9{i11}, iS10{i12}, iS11{i13} ]
   * T7_g[ bS28{1}, iS29{i1}, iS30{i2}, iS31{i3} ];
T5_l[ bS20{1}, iS21{i7}, bS22{1}, bS23{1} ]
   = T1_g[ bS4{1}, iS5{i7}, bS6{1}, bS7{1} ];
T6_g[ bS24{1}, iS25{i7}, bS26{12}, bS27{513} ] = expand( T5_l[ bS20{1}, iS21{i7}, bS22{1}, bS23{1} ], {1, i7, 12, 513} )
T9_l[ bS36{1}, iS37{i7}, iS38{i12}, iS39{i13} ]
   = where(T6_g[ bS24{1}, iS25{i7}, bS26{12}, bS27{513} ]
  , double(0.0000000000000000)
  , T8_g[ bS32{1}, iS33{i11}, iS34{i12}, iS35{i13} ]);
T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ]
   = T9_l[ bS36{1}, iS37{i7}, iS38{i12}, iS39{i13} ]
   * T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ];
T11_g[ bS44{1}, iS45{i7}, iS46{i12}, rS47{i13} ]
   = reduction( T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ], op = add, initial value = double(0.0000000000000000), allreduce = false )
T12_l[ bS48{1}, iS49{i7}, iS50{i12}, bS51{1} ]
   = broadcast( T11_g[ bS44{1}, iS45{i7}, iS46{i12}, rS47{i13} ] )
T13_g[ bS52{1}, iS53{i7}, iS54{i12}, bS55{1} ]
   = T12_l[ bS48{1}, iS49{i7}, iS50{i12}, bS51{1} ];
T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ] = expand( T13_g[ bS52{1}, iS53{i7}, iS54{i12}, bS55{1} ], {1, i7, i12, 513} )
}

g{(pointwise)
inputs:
T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ] float
T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ] float
T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ] float
outputs:
T17_g[ rS68{1}, iS69{i7}, iS70{i12}, iS71{i13} ] float


T15_l[ bS60{1}, iS61{i15}, iS62{i16}, iS63{i17} ]
   = T3_g[ bS12{1}, iS13{i15}, iS14{i16}, iS15{i17} ]
   * T14_g[ bS56{1}, iS57{i7}, iS58{i12}, bS59{513} ];
T16_g[ bS64{1}, iS65{i7}, iS66{i12}, iS67{i13} ]
   = T10_g[ bS40{1}, iS41{i7}, iS42{i12}, iS43{i13} ]
   - T15_l[ bS60{1}, iS61{i15}, iS62{i16}, iS63{i17} ];
T17_g[ rS68{1}, iS69{i7}, iS70{i12}, iS71{i13} ]
   = reduction( T16_g[ bS64{1}, iS65{i7}, iS66{i12}, iS67{i13} ], op = add, initial value = double(0.0000000000000000), allreduce = false )
}

} //Segmented_Fusion
```

So the second part of the pointwise kernel is the one that triggers the scheduler issue.

### Versions

commit a2ce5a68fbae850b31382e5a230aa6c93863b038 (csarofeen/upstream_merge_1011)
Author: jjsjann123 <jiej@nvidia.com>
Date:   Tue Oct 11 17:08:18 2022 -0700

    cherry-pick Patching getitem in partitioner #86713

**Comment by jjsjann123** [comment]
Assigning myself since I'm still working on a cpp repro.

Currently I have
```
TEST_F(NVFuserTest, FusionVectorizeRepro_CUDA) {
  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
  auto fusion = fusion_ptr.get();
  FusionGuard fg(fusion);

  TensorView* tv1 = TensorViewBuilder()
                        .ndims(2)
                        .contiguity({true, true})
                        .build();
  TensorView* tv0 = TensorViewBuilder()
                        .ndims(2)
                        .contiguity({true, true})
                        .build();
  fusion->addInput(tv1);
  fusion->addInput(tv0);

  auto tv7 = sum(tv0, {1}, true);
  auto tv_exp = expand(tv7, {tv0->axis(0)->extent(), IrBuilder::create<Int>(32128)});
  auto tv3 = exp(tv1);
  auto tv8 = mul(tv3, tv_exp);
  auto tv13 = sub(tv0, tv8);
  fusion->addOutput(tv13);

  auto options = at::TensorOptions().dtype(kFloat).device(at::kCUDA, 0);
  at::Tensor t1 = at::empty_strided({4096, 32128}, {32128, 1}, options);
  at::Tensor t0 = at::empty_strided({4096, 32128}, {32128, 1}, options);

  FusionExecutorCache fec(std::move(fusion_ptr));
  auto cg_outputs = fec.runFusionWithInputs({t1, t0});

  auto ref = t0 - t1.exp() * t0.sum((1), true);
  testValidate(fusion, cg_outputs, {t1, t0}, {ref}, __LINE__, __FILE__);
}
```

Which gives me a similar codegen IR but a different error. :cry: 
```
kernel {
T3_l[ bS12{1}, iS13{i1}, iS14{i2}, iS15{i3} ]
   = T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ]
   * T2_g[ bS8{1}, iS9{i11}, iS10{i12}, bS11{513} ];
T4_l[ bS16{1}, iS17{i7}, iS18{i8}, iS19{i9} ]
   = T1_g[ bS4{1}, iS5{i7}, iS6{i8}, iS7{i9} ]
   - T3_l[ bS12{1}, iS13{i1}, iS14{i2}, iS15{i3} ];
T5_g[ rS20{1}, iS21{i7}, iS22{i8}, iS23{i9} ]
   = reduction( T4_l[ bS16{1}, iS17{i7}, iS18{i8}, iS19{i9} ], op = add, initial value = double(0), allreduce = false )

TransformPrinter :
T1_g[ bS4{1}, iS5{i7}, iS6{i8}, iS7{i9} ]
 root domain : (bS4{1},iS5{i7},iS6{i8},iS7{i9})
T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ]
 root domain : (bS0{1},iS1{i1},iS2{i2},iS3{i3})
T2_g[ bS8{1}, iS9{i11}, iS10{i12}, bS11{513} ]
 root domain : (bS8{1},iS9{i11},iS10{i12},bS11{513})
T3_l[ bS12{1}, iS13{i1}, iS14{i2}, iS15{i3} ]
 root domain : (bS12{1},iS13{i1},iS14{i2},iS15{i3})
T4_l[ bS16{1}, iS17{i7}, iS18{i8}, iS19{i9} ]
 root domain : (bS16{1},iS17{i7},iS18{i8},iS19{i9})
T5_g[ rS20{1}, iS21{i7}, iS22{i8}, iS23{i9} ]
 root domain : (rS20{1},iS21{i7},iS22{i8},iS23{i9})
}

[W lower_alias_memory.cpp:568] Warning: Lower_alias_memory : dynamic sized register allocation (function operator())
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i44 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i45 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i46 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i37 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i38 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i39 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i40 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i41 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i42 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i44 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i45 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i46 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i37 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i38 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i39 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i40 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i41 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i42 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i44 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i45 (function isUnrolled)
[W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i46 (function isUnrolled)
unknown file: Failure
C++ exception with description "false INTERNAL ASSERT FAILED at "/opt/pytorch/pytorch/torch/csrc/jit/codegen/cuda/executor.cpp":306, please report a bug to PyTorch. Allocations must be based on constant integers for local memory. However, found: T3_l[ bS12{1}, iS32{T0.size[1]}, iS33{T0.size[2]}, iS34{T0.size[3]} ], T4_l[ bS16{1}, iS35{T0.size[1]}, iS36{T0.size[2]}, iS37{T0.size[3]} ],  have dynamic allocations but are placed in local memory.

```

**Comment by naoyam** [comment]
> Assigning myself since I'm still working on a cpp repro.
> 
> Currently I have
> 
> ```
> TEST_F(NVFuserTest, FusionVectorizeRepro_CUDA) {
>   std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
>   auto fusion = fusion_ptr.get();
>   FusionGuard fg(fusion);
> 
>   TensorView* tv1 = TensorViewBuilder()
>                         .ndims(2)
>                         .contiguity({true, true})
>                         .build();
>   TensorView* tv0 = TensorViewBuilder()
>                         .ndims(2)
>                         .contiguity({true, true})
>                         .build();
>   fusion->addInput(tv1);
>   fusion->addInput(tv0);
> 
>   auto tv7 = sum(tv0, {1}, true);
>   auto tv_exp = expand(tv7, {tv0->axis(0)->extent(), IrBuilder::create<Int>(32128)});
>   auto tv3 = exp(tv1);
>   auto tv8 = mul(tv3, tv_exp);
>   auto tv13 = sub(tv0, tv8);
>   fusion->addOutput(tv13);
> 
>   auto options = at::TensorOptions().dtype(kFloat).device(at::kCUDA, 0);
>   at::Tensor t1 = at::empty_strided({4096, 32128}, {32128, 1}, options);
>   at::Tensor t0 = at::empty_strided({4096, 32128}, {32128, 1}, options);
> 
>   FusionExecutorCache fec(std::move(fusion_ptr));
>   auto cg_outputs = fec.runFusionWithInputs({t1, t0});
> 
>   auto ref = t0 - t1.exp() * t0.sum((1), true);
>   testValidate(fusion, cg_outputs, {t1, t0}, {ref}, __LINE__, __FILE__);
> }
> ```
> 
> Which gives me a similar codegen IR but a different error. 😢
> 
> ```
> kernel {
> T3_l[ bS12{1}, iS13{i1}, iS14{i2}, iS15{i3} ]
>    = T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ]
>    * T2_g[ bS8{1}, iS9{i11}, iS10{i12}, bS11{513} ];
> T4_l[ bS16{1}, iS17{i7}, iS18{i8}, iS19{i9} ]
>    = T1_g[ bS4{1}, iS5{i7}, iS6{i8}, iS7{i9} ]
>    - T3_l[ bS12{1}, iS13{i1}, iS14{i2}, iS15{i3} ];
> T5_g[ rS20{1}, iS21{i7}, iS22{i8}, iS23{i9} ]
>    = reduction( T4_l[ bS16{1}, iS17{i7}, iS18{i8}, iS19{i9} ], op = add, initial value = double(0), allreduce = false )
> 
> TransformPrinter :
> T1_g[ bS4{1}, iS5{i7}, iS6{i8}, iS7{i9} ]
>  root domain : (bS4{1},iS5{i7},iS6{i8},iS7{i9})
> T0_g[ bS0{1}, iS1{i1}, iS2{i2}, iS3{i3} ]
>  root domain : (bS0{1},iS1{i1},iS2{i2},iS3{i3})
> T2_g[ bS8{1}, iS9{i11}, iS10{i12}, bS11{513} ]
>  root domain : (bS8{1},iS9{i11},iS10{i12},bS11{513})
> T3_l[ bS12{1}, iS13{i1}, iS14{i2}, iS15{i3} ]
>  root domain : (bS12{1},iS13{i1},iS14{i2},iS15{i3})
> T4_l[ bS16{1}, iS17{i7}, iS18{i8}, iS19{i9} ]
>  root domain : (bS16{1},iS17{i7},iS18{i8},iS19{i9})
> T5_g[ rS20{1}, iS21{i7}, iS22{i8}, iS23{i9} ]
>  root domain : (rS20{1},iS21{i7},iS22{i8},iS23{i9})
> }
> 
> [W lower_alias_memory.cpp:568] Warning: Lower_alias_memory : dynamic sized register allocation (function operator())
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i44 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i45 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i46 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i37 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i38 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i39 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i40 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i41 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i42 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i44 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i45 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i46 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i37 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i38 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i39 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i40 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i41 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i42 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i44 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i45 (function isUnrolled)
> [W kernel_ir.cpp:395] Warning: Unroll required but not possible. Register allocation disabled. Loop index: i46 (function isUnrolled)
> unknown file: Failure
> C++ exception with description "false INTERNAL ASSERT FAILED at "/opt/pytorch/pytorch/torch/csrc/jit/codegen/cuda/executor.cpp":306, please report a bug to PyTorch. Allocations must be based on constant integers for local memory. However, found: T3_l[ bS12{1}, iS32{T0.size[1]}, iS33{T0.size[2]}, iS34{T0.size[3]} ], T4_l[ bS16{1}, iS35{T0.size[1]}, iS36{T0.size[2]}, iS37{T0.size[3]} ],  have dynamic allocations but are placed in local memory.
> ```

It seems no scheduling is applied to the fusion as there's no transformation shown in the transform printer output.

**Comment by jjsjann123** [comment]
> It seems no scheduling is applied to the fusion as there's no transformation shown in the transform printer output.

I think it currently fails before hitting scheduler. :stuck_out_tongue: Let me try Xiang's fix on this one.

Meanwhile, Kevin also hit the same error with the view_reduction branch and he actually has a repro, which Christian already patched.

**Comment by jjsjann123** [comment]
I'm following Kevin's approach here now and just logging the python repro then.

```
import torch
from torch._C._nvfuser import FusionDefinition, Fusion, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[1, -1, -1, -1], contiguous=[True, True, True, True], dtype=DataType.Bool)
    T1 = fd.define_tensor(symbolic_sizes=[1, -1, 1, 1], contiguous=[True, True, True, True], dtype=DataType.Bool)
    T2 = fd.define_tensor(symbolic_sizes=[1, -1, -1, -1], contiguous=[False, False, False, True], dtype=DataType.Float)
    T3 = fd.define_tensor(symbolic_sizes=[1, -1, -1, -1], contiguous=[True, True, True, True], dtype=DataType.Float)
    T4 = fd.ops.cast(T0, dtype=DataType.Float)
    T5 = fd.ops.broadcast_in_dim(T1, output_shape=[1, 1024, 12, 513], broadcast_dims=[0, 1, 2, 3])
    S6 = fd.define_constant(1.1111111111111112)
    T7 = fd.ops.mul(T4, S6)
    T8 = fd.ops.mul(T2, T7)
    S9 = fd.define_constant(0.0000000000000000)
    T10 = fd.ops.where(T5, S9, T8)
    T11 = fd.ops.mul(T10, T3)
    T12 = fd.ops.sum(T11, axes=[3], keepdim=False, dtype=DataType.Null)
    T13 = fd.ops.broadcast_in_dim(T12, output_shape=[1, 1024, 12, 1], broadcast_dims=[0, 1, 2])
    T14 = fd.ops.broadcast_in_dim(T13, output_shape=[1, 1024, 12, 513], broadcast_dims=[0, 1, 2, 3])
    T15 = fd.ops.mul(T3, T14)
    T16 = fd.ops.sub(T11, T15)
    T17 = fd.ops.squeeze(T16, original_shape=[1, 1024, 12, 513], dim=0)
    fd.add_output(T17)

inputs = [
    torch.randn(1, 1024, 12, 513, device='cuda').bool(),
    torch.randn(1, 1024, 1, 1, device='cuda').bool(),
    torch.randn(1, 1024, 12, 513, device='cuda'),
    torch.randn(1, 1024, 12, 513, device='cuda'),
]

fs = Fusion()
with FusionDefinition(fs) as fd:
    nvfuser_fusion_id1(fd)

for _ in range(5) :
    fs.execute(inputs)

```

This one have been fixed already in Christian's branch https://github.com/csarofeen/pytorch/pull/2062/commits/383c19d7c8936dac45b1df0fb109507dac50cbeb

Just logging things here and will close once that is merged.
